src/Umbraco.Core/Cache/AppCaches.cs

```diff
@@ -86,8 +86,8 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
-                RuntimeCache.DisposeIfDisposable();
-                RequestCache.DisposeIfDisposable();
+                (RuntimeCache as IDisposable)?.Dispose();
```

Comment:
What did you have against `DisposeIfDisposable`?  You could argue it's a useful syntax sugar, meaning you don't have to handle the cast and null check.

Reply:
Ah, I see the `try/catch` in the implementation, so assume it's that you are avoiding right?

Reply:
I'd avoid extension methods on object types to begin with, as that quickly clutters the development experience (as it's added to the auto-completion list of all types, due to being in `Umbraco.Extensions` that's added as global namespace). It also didn't support `IAsyncDisposable`, so instead of adding another extension method, I'd argue being more explicit is the better approach.